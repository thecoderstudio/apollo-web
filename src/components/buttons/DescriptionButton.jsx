import React from 'react';
import OutlinedButton from './OutlinedButton';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { SmallText } from '../Text';

const propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired
};

const StyledOutlinedButton = styled(OutlinedButton)`
  display: grid;
  grid-template-rows: [title] 50px [description] 1fr;
  padding: 20px;
`;

StyledOutlinedButton.displayName = 'OutlinedButton'

const Title = styled.h3`
  grid-row: title;
  width: 80%;
  margin: 0 auto;
  text-align: center;
`;

const Description = styled(SmallText)`
  grid-row: description;
  padding: 10px;
`;

function DescriptionButton(props) {
  return (
    <StyledOutlinedButton>
      <Title>{props.title}</Title>
      <Description>{props.children}</Description>
    </StyledOutlinedButton>
  );
};

DescriptionButton.propTypes = propTypes;

export default DescriptionButton;