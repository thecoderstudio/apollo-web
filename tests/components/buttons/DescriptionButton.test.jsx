import React from 'react';
import renderer from 'react-test-renderer';
import DescriptionButton from '../../../src/components/buttons/DescriptionButton';

function getComponent(props) {
  return renderer.create(
    <DescriptionButton {...props} />
  );
}

describe('DescriptionButton', () => {
  const spy = jest.fn();
  const props = { children: <div />, onClick: spy, title: 'title' };

  it("renders correctly", () => {
    const tree = getComponent(props).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onClick correctly", () => {
    const instance = getComponent({ ...props, loading: true }).root;
    instance.findByType('button').props.onClick();
    expect(spy).toHaveBeenCalled();
  });
});
