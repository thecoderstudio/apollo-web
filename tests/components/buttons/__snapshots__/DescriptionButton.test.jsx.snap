// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionButton renders correctly 1`] = `
.c0 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid;
  display: grid;
  grid-template-rows: [title] 50px [description] 1fr;
  padding: 20px;
  min-height: 180px;
}

.c0:hover {
  opacity: 0.90;
}

.c0:focus {
  outline: none;
}

.c0:active {
  opacity: 0.80;
}

.c0:disabled:hover {
  opacity: 1;
  cursor: auto;
}

.c1 {
  grid-row: title;
  width: 80%;
  margin: 0 auto;
  text-align: center;
}

.c2 {
  grid-row: description;
  padding: 10px;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  <h3
    className="c1"
  >
    title
  </h3>
  <p
    className="c2"
  >
    <div />
  </p>
</button>
`;
